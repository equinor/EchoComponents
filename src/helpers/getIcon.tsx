import React from 'react';
import { ReactComponent as Cable } from '../icons/tags/Cable-icon.svg';
import { ReactComponent as CircuitStarter } from '../icons/tags/Circuit-icon.svg';
import { ReactComponent as ElectricalIcon } from '../icons/tags/Electrical-icon.svg';
import { ReactComponent as Function } from '../icons/tags/Function-icon.svg';
import { ReactComponent as Instrument } from '../icons/tags/Instrument-icon.svg';
import { ReactComponent as JunctionBox } from '../icons/tags/JunctionBox-icon.svg';
import { ReactComponent as Line } from '../icons/tags/Line-icon.svg';
import { ReactComponent as MainEquipment } from '../icons/tags/MainEquipment-icon.svg';
import { ReactComponent as ManualValve } from '../icons/tags/ManualValve-icon.svg';
import { ReactComponent as Signal } from '../icons/tags/Signal-icon.svg';
import { ReactComponent as Tag } from '../icons/tags/Tag-icon.svg';
import { ReactComponent as Telecom } from '../icons/tags/Telecom-icon.svg';
/**
 * Method that returns the relevant Icon for the iconType provided
 * Tag icon is the default JSX.Element Icon returned
 * @export
 * @param {('electrical'
 *         | 'main Equipment'
 *         | 'line'
 *         | 'manual Valve'
 *         | 'circuit/starter'
 *         | 'instrument'
 *         | 'cable'
 *         | 'function'
 *         | 'signal'
 *         | 'telecom'
 *         | 'junction box'
 *         | 'administrative'
 *         | string)} iconType The type of icon you want returned
 * @return {*}  {JSX.Element} Relevant icon for the provided iconType
 */
export function getIcon(
    iconType:
        | 'electrical'
        | 'main equipment'
        | 'line'
        | 'manual valve'
        | 'circuit/starter'
        | 'instrument'
        | 'cable'
        | 'function'
        | 'signal'
        | 'telecom'
        | 'junction box'
        | 'administrative'
        | string
): JSX.Element {
    switch (iconType.toLocaleLowerCase()) {
        case 'electrical':
            return <ElectricalIcon aria-label={'electrical'} />;
        case 'main equipment':
            return <MainEquipment aria-label={'main equipment'} />;
        case 'line':
            return <Line aria-label={'line'} />;
        case 'manual valve':
            return <ManualValve aria-label={'manual valve'} />;
        case 'circuit/starter':
            return <CircuitStarter aria-label={'circuit/starter'} />;
        case 'cable':
            return <Cable title={'cable'} />;
        case 'instrument':
            return <Instrument aria-label={'instrument'} />;
        case 'function':
            return <Function aria-label={'function'} />;
        case 'signal':
            return <Signal aria-label={'signal'} />;
        case 'telecom':
            return <Telecom aria-label={'telecom'} />;
        case 'junction box':
            return <JunctionBox aria-label={'junction box'} />;
        case 'administrative':
            return <Tag aria-label={'tag'} />;
        default:
            return <Tag aria-label={'tag'} />;
    }
}
