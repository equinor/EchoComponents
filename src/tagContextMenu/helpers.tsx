import React from 'react';
import { ReactComponent as Cable } from '../icons/tags/Cable-icon.svg';
import { ReactComponent as CircuitStarter } from '../icons/tags/Circuit-icon.svg';
import { ReactComponent as ElectricalIcon } from '../icons/tags/Electrical-icon.svg';
import { ReactComponent as Function } from '../icons/tags/Function-icon.svg';
import { ReactComponent as Instrument } from '../icons/tags/Instrument-icon.svg';
import { ReactComponent as JunctionBox } from '../icons/tags/JunctionBox-icon.svg';
import { ReactComponent as Line } from '../icons/tags/Line-icon.svg';
import { ReactComponent as MainEquipment } from '../icons/tags/MainEquipment-icon.svg';
import { ReactComponent as ManualValve } from '../icons/tags/ManualValve-icon.svg';
import { ReactComponent as Signal } from '../icons/tags/Signal-icon.svg';
import { ReactComponent as Tag } from '../icons/tags/Tag-icon.svg';
import { ReactComponent as Telecom } from '../icons/tags/Telecom-icon.svg';

export function getIcon(
    iconType:
        | 'Electrical'
        | 'Main Equipment'
        | 'Line'
        | 'Manual Valve'
        | 'Circuit/Starter'
        | 'Instrument'
        | 'Cable'
        | 'Function'
        | 'Signal'
        | 'Telecom'
        | 'Junction Box'
        | 'Administrative'
        | string
): JSX.Element {
    switch (iconType) {
        case 'Electrical':
            return <ElectricalIcon />;
        case 'Main Equipment':
            return <MainEquipment />;
        case 'Line':
            return <Line />;
        case 'Manual Valve':
            return <ManualValve />;
        case 'Circuit/Starter':
            return <CircuitStarter />;
        case 'Cable':
            return <Cable />;
        case 'Instrument':
            return <Instrument />;
        case 'Function':
            return <Function />;
        case 'Signal':
            return <Signal />;
        case 'Telecom':
            return <Telecom />;
        case 'Junction Box':
            return <JunctionBox />;
        case 'Administrative':
            return <Tag />;
        default:
            return <Tag />;
    }
}
