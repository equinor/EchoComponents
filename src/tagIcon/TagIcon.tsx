import React from 'react';
import styles from './tagIcon.module.css';

interface TagIconProps {
    icon: JSX.Element;
    legendColor: string;
}
/**
 * Component that will wrap the provided icon with a background color (usually the legend color)
 *
 * @param {TagIconProps} {
 *  icon: Icon to be wrapped
 *  legendColor: background color to apply. Need to be valid css color
 * }
 * @return {*}  {JSX.Element} Wrapped icon with provided color
 */
export const TagIcon: React.FC<TagIconProps> = ({ icon, legendColor }: TagIconProps): JSX.Element => {
    return (
        <div style={{ background: legendColor }} className={`${styles.icon}`}>
            {icon}
        </div>
    );
};

export default TagIcon;
