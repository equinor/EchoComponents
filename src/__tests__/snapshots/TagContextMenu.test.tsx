import React from 'react';
import renderer from 'react-test-renderer';
import { TagContextMenu } from '../..';
import { getIcon } from '../../helpers/getIcon';
import TagIcon from '../../tagIcon/TagIcon';

it('renders closed TagContextMenu correctly', () => {
    const setExpanded = jest.fn();
    const openTagInformation = jest.fn();

    const tree = renderer
        .create(
            <TagContextMenu
                expanded={false}
                setExpanded={setExpanded}
                tagNo={'11PT100'}
                description={'Pressure Transmitter for Instrument Air Supply '}
                openTagInformation={openTagInformation}
            >
                <TagIcon icon={getIcon('Electrical')} legendColor={'#FFF'} />
            </TagContextMenu>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders open TagContextMenu correctly', () => {
    const setExpanded = jest.fn();
    const openTagInformation = jest.fn();

    const tree = renderer
        .create(
            <TagContextMenu
                expanded={true}
                setExpanded={setExpanded}
                tagNo={'11PT100'}
                description={'Pressure Transmitter for Instrument Air Supply '}
                openTagInformation={openTagInformation}
            >
                <TagIcon icon={getIcon('Electrical')} legendColor={'#FFF'} />
            </TagContextMenu>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
