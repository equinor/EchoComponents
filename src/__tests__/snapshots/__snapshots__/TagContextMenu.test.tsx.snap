// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders closed tagContextMenu correctly 1`] = `
<div
  className="contextWrapper"
  onClick={[Function]}
>
  <div
    className="shadow"
  >
    <div
      className="icon"
      style={
        Object {
          "background": "#FFF",
        }
      }
    >
      <svg />
    </div>
  </div>
</div>
`;

exports[`renders open tagContextMenu correctly 1`] = `
<div
  className="contextWrapperExpanded"
  onClick={[MockFunction]}
>
  <div
    className="tagInfoWrapper"
  >
    <div
      className="icon"
      style={
        Object {
          "background": "#FFF",
        }
      }
    >
      <svg />
    </div>
    <div
      className="tagText"
    >
      <h5
        className="sc-dlfnbm eTggQY tagHeader"
      >
        11PT100
      </h5>
      <p
        className="sc-dlfnbm gsGDwC tagDescription"
      >
        Pressure Transmitter for Instrument Air Supply 
      </p>
    </div>
    <div
      className="tagInfoIcon"
    >
      <button
        className="sc-gsTCUz eUnAE"
        disabled={false}
        tabIndex={0}
        type="button"
      >
        <span
          className="sc-bdfBwQ ibGDsh"
        >
          <svg
            aria-labelledby="eds-info_circle-1"
            className="sc-idOhPF bqIcIz"
            fill="#007079"
            height="24px"
            name={null}
            role="img"
            title="tag information"
            viewBox="0 0 24 24"
            width="24px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <title
              id="eds-info_circle-1"
            >
              tag information
            </title>
            <path
              className="sc-dIUggk"
              clipRule="evenodd"
              d="M12.01 22C17.53 22 22 17.52 22 12S17.53 2 12.01 2C6.48 2 2 6.48 2 12s4.48 10 10.01 10zM13 9V7h-2v2h2zm0 8v-6h-2v6h2zm7-5c0-4.42-3.58-8-8-8s-8 3.58-8 8 3.58 8 8 8 8-3.58 8-8z"
              data-testid="eds-icon-path"
              fillRule="evenodd"
              height="24"
              size={null}
              transform={null}
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
